src/main/java/com/uniovi/util/IncidentSelector.java

```diff
@@ -0,0 +1,92 @@
+package com.uniovi.util;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import com.uniovi.entities.Incident;
+
+public class IncidentSelector {
+
+	private final String confPath="src/main/resources/dataToSave.json";
+	
+	public boolean isRelevant(Incident incident) {
+		for (Function<Incident, Boolean> condition : relevanceConditions()) {
+			if (condition.apply(incident))
+				return true;
+		}
+		return false;
+	}
+
+	private List<Function<Incident, Boolean>> relevanceConditions() {
+		
+		List<Function<Incident, Boolean>> conditions = new ArrayList<Function<Incident, Boolean>>();
+		
+		Reader reader = null;
+		JSONParser parser = new JSONParser();
+
+			try {
+				reader = new FileReader(confPath);
+				JSONObject jsonObject = (JSONObject) parser.parse(reader);
+				for (Object obj : jsonObject.keySet()) {
+					obtainCondition(conditions, jsonObject, (String) obj);
+				}
+			} catch (FileNotFoundException e) {
+				e.printStackTrace();
+			} catch (IOException e) {
+				e.printStackTrace();
+			} catch (ParseException e) {
+				e.printStackTrace();
+			}
+		return conditions;
+	}
+
+	private void obtainCondition(List<Function<Incident, Boolean>> conditions, JSONObject jsonObject, String key) {
+		if (key.equals("kind"))
+			conditionsKind(conditions, jsonObject);
+		else if (key.equals("temperature"))
+			conditionsSensor(conditions, jsonObject, key);
+		else if (key.equals("pollution"))
+			conditionsSensor(conditions, jsonObject, key);
+	}
+	
+	private void conditionsKind(List<Function<Incident, Boolean>> conditions, JSONObject jsonObject) {
+		
+		JSONObject kindContents = (JSONObject) jsonObject.get("kind");
+		if (kindContents.get("type").equals("values")){
+			JSONArray kinds = (JSONArray) kindContents.get("important");
+			for (int j = 0; j < kinds.size(); j++) {
+				Integer index = j;
+				Function<Incident, Boolean> function = i -> i.getAgent().getKind().equals(kinds.get(index));
+				conditions.add(function);
+			}
+		}
+	}
+	
+	private void conditionsSensor(List<Function<Incident, Boolean>> conditions, JSONObject jsonObject, String sensorType) {
+		JSONObject jsonContents = (JSONObject) jsonObject.get(sensorType);
+		if (jsonContents.get("type").equals("range"))
+			conditionRange(sensorType, conditions, jsonContents);		
+	}
+	
+	private void conditionRange(String sensorType, List<Function<Incident, Boolean>> conditions, JSONObject jsonObject) {
+		Long min =  (Long) jsonObject.get("min");
+		Long max =  (Long) jsonObject.get("max");
+		
+		Function<Incident, Boolean> func = 
+				i -> i.getTags().contains(sensorType) 
+						&& (((Double) i.getProperties().get("value")) < min 
```

Comment:
As I understand your idea is to have incidents with this format:
```json
{
   "inciName": "whatever",
   "agent": "...",
   "tags": ["pollution"],
   "properties": {
      "value": x,
   }
}
```
The problem with this approach is that you can't have for example a sensor measuring several values (temperature, pollution...), right?

My idea was having something like this:
```json
{
   "inciName": "whatever",
   "agent": "...",
   "tags": ["whatever"],
   "properties": {
      "pollution": x,
      "temperature": y,
   }
}
```
So you can have many different values in the properties map. Let me know what you think about this approach.

Reply:
I think something like this should work:
```java
i -> i.getProperties().containsKey(sensorType)  && (((Double) i.getProperties().get(sensorType)) < min && ...)) 
```

Reply:
I just assumed each sensor would measure a single thing, so it made sense to identify what a sensor did in the tags. I get what you mean though, as meteo sensors would probably send in a single incident a lot of information (like temperature, rain, pressure...). 

I will change that in a bit.

Reply:
It's not well defined in the requirements how agents send the information. As you said, including the information in the tags was a possibility, while my idea was storing it in the properties. You can do as you wish, but I think using the properties gives us more flexibility.